api/endpoint/staking.go

```diff
@@ -96,6 +97,51 @@ func GetStakeInfoForBatch(c *gin.Context, apis map[string]blockatlas.StakeAPI) {
 	c.JSON(http.StatusOK, blockatlas.DocsResponse{Docs: &batch})
 }
 
+// @Summary Get staking info by coin ID
+// @ID batch_info
+// @Description Get staking info by coin ID
+// @Produce json
+// @Tags Staking
+// @Param coins query string true "List of coins"
+// @Success 200 {array} blockatlas.DelegationsBatchPage
+// @Failure 400 {object} model.ErrorResponse
+// @Router /v3/staking/list [get]
+func GetStakeInfoForCoins(c *gin.Context, apis map[string]blockatlas.StakeAPI) {
+	coinsRequest := c.Query("coins")
+	if coinsRequest == "" {
+		c.JSON(http.StatusBadRequest, model.CreateErrorResponse(model.InvalidQuery, errors.E("empty coins list")))
+		return
+	}
+
+	coinsRaw := strings.Split(coinsRequest, ",")
+
+	coins, err := sliceAtoi(coinsRaw)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, model.CreateErrorResponse(model.InvalidQuery, err))
+		return
+	}
+
+	var reqs CoinsRequest
+	for _, c := range coins {
+		reqs = append(reqs, CoinBatchRequest{Coin: uint(c)})
+	}
+
+	batch := make(blockatlas.StakingBatchPage, 0)
+	for _, r := range reqs {
+		requestCoin, ok := coin.Coins[r.Coin]
+		if !ok {
+			continue
+		}
+		p, ok := apis[requestCoin.Handle]
+		if !ok {
```

Comment:
any reason to duplicate this code: https://github.com/trustwallet/blockatlas/blob/48e9063a65b7e03af85761152fd70d356c05bd18/api/endpoint/staking.go#L129 ?